### ISSUE TITLE
pg_upgrade fails on clusters that ran  6.1 or below

### ISSUE THREAD

**Issue by fdr** [citusdata/citus#1788]
Seems at some point the function `master_get_round_robin_candidate_nodes` was removed. The only extension control file to reflect this is `citus--6.1-17--6.2-1.sql`.

However, it fails to drop the function, and succeeds, because it is written like this:

```sql
DROP FUNCTION IF EXISTS master_get_round_robin_candidate_nodes();
```

This statement succeeds, because there is no such function signature, nor has there ever been. What *does* exist is the function with the signature `master_get_round_robin_candidate_nodes(bigint)`, which lives on after the control file has run.

To confirm this, on a 7.0 installation, if one tries to call this function, predictably, it fails:

```
ERROR:  could not find function "master_get_round_robin_candidate_nodes" in file "/usr/pgsql-9.6/lib/citus.so"
```

That's fine, as it is dead code...until you try to pg_upgrade.

pg_upgrade attempts to resolve every symbol of LANGUAGE C functions in their corresponding .sos. And that fails, here, and the result is pg_upgrade exits with a failure.

Suggested actions:

* throw in a stub symbol/function into new point releases of the .sos that simply satisfies pg_upgrade, and otherwise does nothing.

* drop the function correctly in migrations to new versions so the stub can eventually go away.

**Comment by sumedhpathak** [comment]
@fdr To clarify then, this should also have manifested itself in a 6.1/PG9.5 upgrade to 6.2/PG9.6?

**Comment by fdr** [comment]
I am not sure what you are asking, because one doesn't do a postgres and citus version upgrade at precisely the same time. What is true is 6.2/9.5 to 6.2/9.6 should exhibit it. The 6.1 or 6.0 variants of these upgrades are thought to be fine as the function hasn't been removed yet.

**Comment by marcocitus** [comment]
> throw in a stub symbol/function into new point releases of the .sos that simply satisfies pg_upgrade, and otherwise does nothing.

Makes sense. Do you need this backported to 7.0 so can perform major upgrades on 7.0? Or is 7.1 ok?

**Comment by fdr** [comment]
Backport would help. If you only add the fix to 7.1 you may as well throw it into the upgrade sql and not bother with the stub.
